application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep

```diff
@@ -0,0 +1,412 @@
+var userData = 'team_id=${parsec_teamId}:key=${parsec_teamKey}:name=${parsec_host}:user_email=${parsec_userEmail}:is_guest_access=${parsec_isGuestAccess}:ibLicenseKey=${ibLicenseKey}'
+var Script2Run = 'TeradiciRegCAS.ps1'
+var CSEParams = ' -pcoip_registration_code ${teradiciRegKey}'
+var cmdTeradiciRegistration = './${Script2Run}${CSEParams}'
+
+var vnetId = {
+  'new'     : resourceId('Microsoft.Network/virtualNetworks', vnetName)
+  'existing': resourceId(vnetRGName, 'Microsoft.Network/virtualNetworks', vnetName)
+}
+var subnetId = '${vnetId[vnetNewOrExisting]}/subnets/${subNetName}'
+
+var publicIpId = {
+  'new': resourceId('Microsoft.Network/publicIPAddresses', publicIpName)
+  'existing': resourceId(publicIpRGName, 'Microsoft.Network/publicIPAddresses', publicIpName) 
+  'none': ''
+}[publicIpNewOrExisting]
+
+resource partnercenter 'Microsoft.Resources/deployments@2021-04-01' = {
+  name: 'pid-7837dd60-4ba8-419a-a26f-237bbe170773-partnercenter'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource publicIp 'Microsoft.Network/publicIPAddresses@2021-05-01' = if (publicIpNewOrExisting == 'new') {
+  name: publicIpName
+  location: location
+  sku: {
+    name: publicIpSku
+  }
+  properties: {
+    publicIPAllocationMethod: publicIpAllocationMethod
+    dnsSettings: {
+      domainNameLabel: publicIpDns
+    }
+  }
+}
+
+module nsg_rules 'nsgRules.bicep' = {
+  name: 'nsg_rules'
+  params: {
+    addPixelStreamingPorts: unrealPixelStreamingEnabled
+  }
+}
+
+resource nsg 'Microsoft.Network/networkSecurityGroups@2021-05-01' = {
+  name: nsgName
+  location: location
+  properties: {
+    securityRules: nsg_rules.outputs.nsgRules['nsgRules-${remoteAccessTechnology}']
+  }
+}
+
+resource vnet 'Microsoft.Network/virtualNetworks@2021-05-01' = if (vnetNewOrExisting == 'new') {
+  name: vnetName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        first(vnetARPrefixes)
+      ]
+    }
+    subnets: [
+      {
+        name: subNetName
+        properties: {
+          addressPrefix: subNetARPrefix
+        }
+      }
+    ]
+  }
+}
+
+resource nic 'Microsoft.Network/networkInterfaces@2021-05-01' = {
+  name: nicName
+  location: location
+  dependsOn: [
+    vnet
+  ]
+  properties: {
+    enableAcceleratedNetworking: (bool(length(split(vmSize, '_')) > 2) ? true : false)
+    ipConfigurations: [
+      {
+        name: ipconfName
+        properties: union( {
+              subnet: { 
+                id: subnetId
+              }
+            }, { 
+              privateIPAllocationMethod: 'Dynamic' 
+            }, (!empty(publicIpId)) ? { 
```

Comment:
will need to change this logic a bit when you flatten the var above

Reply:
did not have to modify this... (at least i dont think so...)